src/devices/Card/CreditCard/CreditCard.cs

```diff
@@ -0,0 +1,497 @@
+                        var dol = new Tag();
+                        dol.Data = new byte[1];
+                        if ((dolList[index] & 0b0001_1111) == 0b0001_1111)
+                        {
+                            dol.TagNumber = BinaryPrimitives.ReadUInt16BigEndian(dolList.AsSpan().Slice(index, 2));
+                            index += 2;
+                            dolList.AsSpan().Slice(index++, 1).CopyTo(dol.Data);
+                        }
+                        else
+                        {
+                            dol.TagNumber = dolList[index++];
+                            dolList.AsSpan().Slice(index++, 1).CopyTo(dol.Data);
+                        }
+                        dol.Parent = tag.TagNumber;
+                        tag.Tags.Add(dol);
+                    }
+                }
+                tag.Parent = parent;
+                theTags.Add(tag);
+            }
+
+        }
+
+        /// <summary>
+        /// Gather all the public information present in the credit card.
+        /// Fill then Tag list with all the found information
+        /// </summary>
+        public void FillCreditCardInformation()
+        {
+            // This is a string "2PAY.SYS.DDF01" (PPSE) to select the root directory
+            var ret = Select(RootDirectory);
+            if (ret == ErrorType.ProcessCompletedNormal)
+            {
+                // Find all Application Template = 0x61
+                var appTemplates = Tag.SearchTag(Tags, 0x61);
+                LogInfo.Log($"Number of App Templates: {appTemplates?.Count}", LogLevel.Debug);
+                foreach (var app in appTemplates)
+                {
+                    // Find the Application Identifier 0x4F
+                    var appIdentifier = Tag.SearchTag(app.Tags, 0x4F).FirstOrDefault();
+                    // Find the Priority Identifier 0x87
+                    var appPriotity = Tag.SearchTag(app.Tags, 0x87).FirstOrDefault();
+                    // As it is not mandatory, some cards will have only 1
+                    // application and this may not be present
+                    if (appPriotity == null)
+                        appPriotity = new Tag() { Data = new byte[1] { 0 } };
+                    // do we have a PDOL tag 0x9F38 
+
+                    LogInfo.Log($"APPID: {BitConverter.ToString(appIdentifier.Data)}, Priority: {appPriotity.Data[0]}", LogLevel.Debug);
+                    ret = Select(appIdentifier.Data);
+                    if (ret == ErrorType.ProcessCompletedNormal)
+                    {
+
+                        // We need to select the Template 0x6F where the Tag 0x84 contains the same App Id and where we have a template A5 attached. 
+                        var templates = Tags.Where(m => m.TagNumber == 0x6F).Where(m => m.Tags.Where(t => t.TagNumber == 0x84).Where(t => t.Data.SequenceEqual(appIdentifier.Data)) != null).Where(m => m.Tags.Where(t => t.TagNumber == 0xA5) != null);
```

Comment:
huh? at minimum split or format it in such way that it's clear what's executed in what order

Reply:
I know this one is a bit complicated that's the reason why I added the comment. I formated it differently so hopefully it will make it a bit more clear